File: java/DesignPatterns/CreationalDesignPatterns/FactoryDesignPattern/FactoryDesignPattern.Example1/src/main/java/creational/factory/example1/services/concretes/SmsNotificationService.java
package creational.factory.example1.services.concretes;

import creational.factory.example1.services.abstracts.INotificationService;

public class SmsNotificationService implements INotificationService {
    public String send(String to, String message) {
        String result = "Sending SMS to " + to + " with message: " + message;
        System.out.println(result);
        return result;
    }
}

File: nodejs/DesignPatterns/CreationalDesignPatterns/FactoryDesignPattern/FactoryDesignPattern.Example1/src/services/SmsNotificationService.ts
import { INotificationService } from './INotificationService';

export class SmsNotificationService implements INotificationService {
  send(to: string, message: string): string {
    const result = `Sending SMS to ${to} with message: ${message}`;
    console.log(result);
    return result;
  }
}
